source/operations/concepts/architecture.rst

```diff
@@ -0,0 +1,189 @@
+   For a more complete discussion of MinIO redundancy and healing, see :ref:`minio-erasure-coding`.
+
+   .. figure:: /images/architecture/architecture-erasure-set-shard.svg
+      :figwidth: 100%
+      :alt: Diagram of object being sharded into 4 data and 4 parity blocks, distributed across 8 drives
+      :align: center
+
+      MinIO shards the object into 4 data and 4 parity blocks, distributing them across the drives in the erasure set. 
+
+MinIO uses a deterministic algorithm to select the erasure set for a given object.
+   For each unique object namespace ``BUCKET/PREFIX/[PREFIX/...]/OBJECT.EXTENSION``, MinIO always selects the same erasure set for read/write operations.
+   MinIO handles all routing within pools and erasure sets, making the select/read/write process entirely transparent to applications.
+
+   .. figure:: /images/architecture/architecture-erasure-set-retrieve-object.svg
+      :figwidth: 100%
+      :alt: Diagram of object retrieval from only data shards
+      :align: center
+
+      MinIO reconstructs objects from data or parity shards transparently before returning the object to the requesting client.
+
+Each MinIO server has a complete picture of the distributed topology, such that an application can connect and direct operations against any node in the deployment.
+   The MinIO node automatically handles routing internal requests to other nodes in the deployment *and* returning the final response to the client.
+
+   Applications typically should not manage those connections, as any changes to the deployment topology would require application updates.
+   Production environments should instead deploy a load balancer or similar network control plane component to manage connections to the MinIO deployment.
+   For example, you can deploy an NGINX load balancer to perform "least connections" or "round robin" load balancing against the available nodes in the deployment.
+
+   .. figure:: /images/architecture/architecture-load-balancer-8-node.svg
+      :figwidth: 100%
+      :alt: Diagram of an 8-node MinIO deployment behind a load balancer
+      :align: center
+
+      The load balancer routes the request to any node in the deployment.
+      The node which receives the requests handles any internode requests thereafter.
+
+You can expand a MinIO deployment's available storage through :ref:`pool expansion <expand-minio-distributed>`.
+   Each pool consists of an independent group of nodes with their own erasure sets.
+   MinIO must query each pool to determine the correct erasure set to which it directs read and write operations, such that each additional pool adds increased internode traffic per call.
+   The pool which contains the correct erasure set then responds to the operation, remaining entirely transparent to the application.
+
+   If you modify the MinIO topology through pool expansion, you can update your applications by modifying the load balancer to include the new nodes.
```

Comment:
I'm not entirely following this. Is `pool expansion` new nodes in one pool or additional pools?